### PR TITLE
LocaltuyaCover now inherits from LocalTuyaEntity

### PR DESCRIPTION
**Pull request by rospogrigio**
Applied #18 to cover platform too.
@postlund can you review this please? Code is getting sleeker and sleeker... 😄

### CODE REVIEW

**Comment by postlund** [comment]
I will have a look later today 👍

File: custom_components/localtuya/__init__.py [hunk 1]
```diff
@@ -112,6 +112,19 @@ def __init__(self, device, config_entry, dps_id, **kwargs):
         self._dps_id = dps_id
         self._status = None
 
+    @property
+    def device_info(self):
+        return {
+            "identifiers": {
+                # Serial numbers are unique identifiers within a specific domain
+                ("LocalTuya", f"local_{self._device.unique_id}")
```
Comment (postlund) [inline custom_components/localtuya/__init__.py:120]: I believe `"LocalTuya"` should be `DOMAIN`.

File: custom_components/localtuya/__init__.py [hunk 2]
```diff
@@ -112,6 +112,19 @@ def __init__(self, device, config_entry, dps_id, **kwargs):
...
+            },
+            "name": self._device._friendly_name,
+            "manufacturer": "Unknown",
+            "model": "Tuya generic",
+            "sw_version": "3.3",
```
Comment (postlund) [inline custom_components/localtuya/__init__.py:125]: We can use version from config entry:

```
self._config_entry.data[CONF_PROTOCOL_VERSION]
```

File: custom_components/localtuya/__init__.py [hunk 3]
```diff
@@ -112,6 +112,19 @@ def __init__(self, device, config_entry, dps_id, **kwargs):
...
         self._status = None
 
+    @property
+    def device_info(self):
+        return {
+            "identifiers": {
+                # Serial numbers are unique identifiers within a specific domain
+                ("LocalTuya", f"local_{self._device.unique_id}")
+            },
+            "name": self._device._friendly_name,
```
Comment (postlund) [inline custom_components/localtuya/__init__.py:122]: Friendly name is in config entry:

```
self._config_entry.data[CONF_FRIENDLY_NAME]
```
And revert change in the cache until we remove it.

**Review by postlund — CHANGES_REQUESTED** [review]
Minor things we can fix in `device_info` now.

File: custom_components/localtuya/__init__.py (hunk 2 above)
Comment (rospogrigio) [reply custom_components/localtuya/__init__.py:125]: Yep, that's what I wanted to do soon.

**Comment by rospogrigio** [comment]
Anything else? Otherwise I'll merge this.

**Comment by postlund** [comment]
Ok to merge! 👍

**Comment by postlund** [comment]
Feel free to do other remaining platforms if you feel like it 😉

**Comment by rospogrigio** [comment]
> Feel free to do other remaining platforms if you feel like it 😉

Would do it with pleasure but I don't own those devices so I cannot test them...

**Comment by postlund** [comment]
Same here. Shouldn't be that hard to make right though. At some point we should add tests...

**Pull request merged**